### PR TITLE
fix(entity): delete incoming relationships when deleting an entity #162

### PR DESCRIPTION
**Pull request by bobeal**

### CODE REVIEW

File: entity-service/src/main/resources/neo4j/migrations/V06__remove_dangling_relationships.cypher [hunk 1]
```diff
@@ -0,0 +1,3 @@
+MATCH (rel:Relationship)
```
Comment (HoucemKacem) [inline entity-service/src/main/resources/neo4j/migrations/V06__remove_dangling_relationships.cypher:1]: I'm not totally sure about this migration script.
The fact that we match all relationships and we delete those not pointing to Entity, are we sure that all relationships in the graph (should) target entities ?  
Also it fails in local with ConstraintValidationFailed error.

Comment (bobeal) [reply entity-service/src/main/resources/neo4j/migrations/V06__remove_dangling_relationships.cypher:1]: Hmm, yes, a relationship node always targets an entity (that's the NGSI-LD data model), are you thinking about something specific?

I gonna check for the constraint error.

Comment (bobeal) [reply entity-service/src/main/resources/neo4j/migrations/V06__remove_dangling_relationships.cypher:1]: Damn, partial entities, I forgot about them...

Comment (bobeal) [reply entity-service/src/main/resources/neo4j/migrations/V06__remove_dangling_relationships.cypher:1]: OK, updated

Comment (HoucemKacem) [reply entity-service/src/main/resources/neo4j/migrations/V06__remove_dangling_relationships.cypher:1]: Yes I was thinking about Partial Entities.
LGTM

**Pull request merged**